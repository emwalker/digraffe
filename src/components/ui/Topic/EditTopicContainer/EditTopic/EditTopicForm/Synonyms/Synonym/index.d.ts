import { Component, ReactNode } from 'react';
import { Synonym_synonym as SynonymType } from '__generated__/Synonym_synonym.graphql';
declare type Props = {
    dragHandle?: ReactNode;
    onDelete?: (index: number) => void;
    position?: number;
    synonym: SynonymType;
};
declare class Synonym extends Component<Props> {
    static defaultProps: {
        onDelete: undefined;
    };
    onClick: () => void;
    renderDeleteButton: () => JSX.Element;
    render: () => JSX.Element;
}
export declare const UnwrappedSynonym: typeof Synonym;
declare const _default: import("react-relay").Container<Props>;
export default _default;
