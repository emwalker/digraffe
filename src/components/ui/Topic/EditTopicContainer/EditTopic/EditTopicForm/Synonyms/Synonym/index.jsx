// @flow
import React, { Component } from 'react'
import Octicon from 'react-component-octicons'
import { createFragmentContainer, graphql } from 'react-relay'

import type { Synonym_synonym as SynonymType } from './__generated__/Synonym_synonym.graphql'

type Props = {
  onDelete?: ?Function,
  // $FlowFixMe
  synonym: SynonymType,
}

class Synonym extends Component<Props> {
  static defaultProps = {
    onDelete: null,
  }

  onClick = () => {
    if (!this.props.onDelete) return
    this.props.onDelete(this.props.synonym)
  }

  renderDeleteButton = () => (
    // eslint-disable-next-line jsx-a11y/click-events-have-key-events
    <span tabIndex="0" role="button" onClick={this.onClick}>
      <Octicon name="trashcan" />
    </span>
  )

  render = () => (
    <li className="Box-row clearfix css-truncate p-2">
      <div className="col-10 float-left">{this.props.synonym.name}</div>
      <div className="col-1 float-right remove-synonym">
        { this.props.onDelete && this.renderDeleteButton() }
      </div>
      <div className="col-1 float-right">
        { this.props.synonym.locale }
      </div>
    </li>
  )
}

export const UnwrappedSynonym = Synonym

export default createFragmentContainer(Synonym, {
  synonym: graphql`
    fragment Synonym_synonym on Synonym {
      id
      name
      locale
    }
  `,
})
