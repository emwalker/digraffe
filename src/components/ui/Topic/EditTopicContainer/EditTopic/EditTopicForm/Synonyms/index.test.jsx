import React from 'react'
import { shallow } from 'enzyme'

import { UnwrappedSynonyms } from './index'
import Synonym from './Synonym'

const synonymEdge = {
  node: {
    name: 'Gnusto',
    id: '1234',
  },
}

const topic = {
  viewerCanDeleteSynonyms: true,
  synonyms: {
    edges: [synonymEdge],
  },
}

const props = {
  topic,
}

describe('<Synonyms />', () => {
  const wrapper = shallow(<UnwrappedSynonyms {...props} />)
  const rows = () => wrapper.find(Synonym)

  it('renders', () => {
    expect(wrapper).toMatchSnapshot()
  })

  describe('when viewerCanDeleteSynonyms: false', () => {
    beforeEach(() => {
      wrapper.setProps({ topic: { ...topic, viewerCanDeleteSynonym: false } })
    })

    it('does not provide an onDelete handler', () => {
      expect(rows().at(0).prop('onDelete')).toBeFalsy()
    })
  })

  describe('when viewerCanDeleteSynonyms: true', () => {
    beforeEach(() => {
      wrapper.setProps({ topic: { ...topic, viewerCanDeleteSynonym: true } })
    })

    it('provides an onDelete handler', () => {
      expect(rows().at(0).prop('onDelete')).toBeTruthy()
    })
  })
})
