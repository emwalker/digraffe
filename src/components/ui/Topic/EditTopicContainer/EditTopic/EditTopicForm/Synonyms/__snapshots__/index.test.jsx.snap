// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Synonyms /> renders 1`] = `
<dl
  className="form-group"
>
  <label
    htmlFor="names-and-synonyms"
  >
    Names and synonyms
  </label>
  <ul
    className="Box list-style-none mt-1 mb-2"
  >
    <SynonymList
      onAdd={[Function]}
      onDelete={[Function]}
      onUpdate={[Function]}
      synonyms={
        Object {
          "edges": Array [
            Object {
              "node": Object {
                "id": "1234",
                "name": "Gnusto",
              },
            },
          ],
        }
      }
      topic={
        Object {
          "synonyms": Object {
            "edges": Array [
              Object {
                "node": Object {
                  "id": "1234",
                  "name": "Gnusto",
                },
              },
            ],
          },
          "viewerCanDeleteSynonyms": true,
        }
      }
    />
  </ul>
  <div
    className="clearfix"
  >
    <input
      className="form-control col-10 mr-2"
      id="names-and-synonyms"
      onChange={[Function]}
      value=""
    />
    <select
      className="form-select col-1 mr-2"
      onChange={[Function]}
    >
      <option>
        en
      </option>
      <option>
        es
      </option>
      <option>
        fr
      </option>
    </select>
  </div>
</dl>
`;
